src/os/bluestore/BlueStore.cc

```diff
@@ -7285,6 +7284,30 @@ BlueStore::OnodeRef BlueStore::fsck_check_objects_shallow(
         (*pu) |= (1u << i);
       }
     }
+    if (depth != FSCK_SHALLOW && blob.has_unused()) {
+      ceph_assert(referenced);
+      auto p = referenced->find(l.blob);
+      bluestore_blob_t::unused_t *pu;
+      if (p == referenced->end()) {
+        pu = &(*referenced)[l.blob];
+      } else {
+        pu = &p->second;
+      }
+      uint64_t unused_len = blob.get_unused_len();
```

Comment:
May be consider moving unused value calculation to bluestore_blob_t struct for both this and legacy value calculation?